library/templates/address_list_form.php

```diff
@@ -39,208 +39,223 @@
-                        ,'field_id' => $field_id_esc . "[period_end][]"
-                        ,'smallform' => ($smallform ?? '') . ' form_addresses_period_end'
-                    ], '');
-                    ?>
-                </div>
+            <div class="col-6 px-0">
+                <span class="display_addresses_full_address label_custom" style="vertical-align: 0;"></span>
             </div>
 
-            <!-- Header -->
-            <div class="row">
-                <div class="col-3">
-                    <?php echo xlt("Address"); ?>
-                </div>
-                <div class="col-3">
-                    <?php echo xlt("Address Line 2"); ?>
-                </div>
-                <div class="col-3">
-                    <?php echo xlt("City"); ?>
-                </div>
-                <div class="col-3">
-                    <?php echo xlt("County/District"); ?>
-                </div>
+            <div class="col-3 display_addresses_period_column px-0">
+                <span class="display_addresses_period label_custom" style="vertical-align: 0;"></span>
             </div>
 
-            <!-- VALUES -->
-            <div class="row">
-                <div class="col-3">
-                    <?php
-                    generate_form_field([
-                        'data_type' => $widgetConstants['textbox']
-                        ,'field_id' => $field_id_esc . "[line_1][]"
-                        ,'smallform' => ($smallform ?? '') . ' form_addresses_line1'
-                    ], '');
-                    ?>
-                </div>
-                <div class="col-3">
-                    <?php
-                    generate_form_field([
-                        'data_type' => $widgetConstants['textbox']
-                        ,'field_id' => $field_id_esc . "[line_2][]"
-                        ,'smallform' => ($smallform ?? '') . ' form_addresses_line2'
-                    ], '');
-                    ?>
-                </div>
-                <div class="col-3">
-                    <?php
-                    generate_form_field([
-                        'data_type' => $widgetConstants['textbox']
-                        ,'field_id' => $field_id_esc . "[city][]"
-                        ,'smallform' => ($smallform ?? '') . ' form_addresses_city'
-                    ], '');
-                    ?>
-                </div>
-                <div class="col-3">
-                    <?php
-                    generate_form_field([
-                        'data_type' => $widgetConstants['textbox']
-                        ,'field_id' => $field_id_esc . "[district][]"
-                        ,'smallform' => ($smallform ?? '') . ' form_addresses_district'
-                    ], '');
-                    ?>
-                </div>
+            <div class="fas fa-fw fa-trash-alt text-danger btn-delete-address" style="text-align: center; flex: 0 0 2em; line-height: 1.5;">
```

Comment:
Need to give this tag a role so we get a cursor pointer for the fake button
`<div class="fas fa-fw fa-trash-alt text-danger btn-delete-address" role="button" style="text-align: center; flex: 0 0 2em; line-height: 1.5;">`
This could easily be done as below in keeping with themes
`<div class="btn btn-delete text-danger p-0 m-0 btn-delete-address">`
![image](https://user-images.githubusercontent.com/1859985/178329288-6ed23187-81e0-400d-8fa9-aca50337ec07.png)

I don't understand why we need inline styles for alignments. For one I question why we are vertical aligning to top by default.(Not on you David). All these alignments can be done in BS classes and should have been sorted out when we refactored demographic.

Soon I'll be going through all this when I refactor out table views.